lib/application.js

```diff
@@ -102,6 +103,9 @@ module.exports = class Application extends Emitter {
    */
 
   use(fn) {
+    if (this.forceAsync && fn.constructor.name !== 'AsyncFunction') {
+      throw new TypeError('middleware must be an AsyncFunction!');
+    }
```

Comment:
If this does land, seems like this check should go after the following one (`if (typeof fn !== 'function') ...`) so that accidentally passing a non-function gives a more helpful error.